infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -97,25 +100,76 @@ public LogReplicationMetadataManager(CorfuRuntime runtime, LogReplicationContext
-
-            this.replicationEventTable = this.corfuStore.openTable(NAMESPACE, REPLICATION_EVENT_TABLE_NAME,
-                    ReplicationEventInfoKey.class,
-                    ReplicationEvent.class,
-                    null,
-                    TableOptions.fromProtoSchema(ReplicationEvent.class));
-        } catch (Exception e) {
-            log.error("Caught an exception while opening metadata tables", e);
-            throw new ReplicationWriterException(e);
+            table = corfuStore.openTable(namespace, tableName,
+                    keyClass, valueClass, metadataClass,
+                    TableOptions.fromProtoSchema(valueClass));
+        } catch (Exception ee) {
+            log.error("Caught an exception while opening "+namespace+"$"+tableName, ee);
+            throw new ReplicationWriterException(ee);
         }
+        return table;
+    }
+
+    /**
+     * The following tables need to move from old type to new type
+     * 1. CORFU-REPLICATION-WRITER-UUID-OF-CLUSTER -> LogReplicationMetadataTable
+     * Type: Schema change AND name change
+     *    (LogReplicationMetadataKey, LogReplicationMetadataVal) ->
+     *     (LogReplicationSession, ReplicationMetadata)
+     * Action: Open the old table with old types and Clear the old table
+     *         Not used in new code. Optional as failure to clear won't cause issues.
+     *         TODO: consider scanning registry for signature and wipe out contents.
+     *
+     * 2. LogReplicationStatus
+     * Type: Schema change for key AND value types with NO table name change
+     * (ReplicationStatusKey, ReplicationStatusVal) ->
+     * (LogReplicationSession.class, ReplicationMetadata.class)
+     * Action: Make serializer aware of old types, open table with new type, clear it
+     *
+     * 3. LogReplicationEventTable
+     * Type: Schema change for Key type only
+     * (ReplicationEventKey, ReplicationEvent) ->
+     * (ReplicationEventInfoKey, ReplicationEvent)
+     * Action: Make serializer aware of old key type, open table with new type, clear it
+     * @param corfuStore - the same runtime used by LogReplicationMetadataManager
+     */
+    public static void addLegacyTypesToSerializer(CorfuStore corfuStore) {
+        corfuStore.getRuntime().getTableRegistry()
+                .addTypeToClassMap(LogReplicationMetadata.ReplicationStatusKey.getDefaultInstance());
+        corfuStore.getRuntime().getTableRegistry()
+                .addTypeToClassMap(LogReplicationMetadata.ReplicationStatusVal.getDefaultInstance());
+        corfuStore.getRuntime().getTableRegistry()
+                .addTypeToClassMap(LogReplicationMetadata.ReplicationEventKey.getDefaultInstance());
+    }
+
+    public static void migrateData(TxnContext txnContext) {
+        txnContext.clear(txnContext.getTable(LogReplicationUtils.REPLICATION_STATUS_TABLE_NAME));
+        txnContext.clear(txnContext.getTable(REPLICATION_EVENT_TABLE_NAME));
```

Comment:
quick question Sundar, looks like you're considering replication_status and replication_event table. Why METADATA_TABLE_NAME is not involved here?

Reply:
the METADATA_TABLE_NAME did not exist in prior version, so it does not need any migration. we only have to handle migration for tables where the table name is the same but the schema under the hood has changed.